client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/GlobalPrivileges.java

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * Represents global privileges. "Global Privilege" is a mantra for granular
+ * privileges over applications. {@code ApplicationResourcePrivileges} model
+ * application privileges over resources. This models user privileges over
+ * applications. Every client is responsible to manage the applications as well
+ * as the privileges for them.
+ */
+public final class GlobalPrivileges implements ToXContentObject {
+
+    static final ParseField APPLICATION = new ParseField("application");
+
+    @SuppressWarnings("unchecked")
+    static final ConstructingObjectParser<GlobalPrivileges, Void> PARSER = new ConstructingObjectParser<>("global_application_privileges",
+            false, constructorObjects -> {
+                // ignore_unknown_fields is irrelevant here anyway, but let's keep it to false
+                // because this conveys strictness (woop woop)
+                return new GlobalPrivileges((Collection<GlobalScopedPrivilege>) constructorObjects[0]);
+            });
+
+    static {
+        PARSER.declareNamedObjects(optionalConstructorArg(), (p, c, n) -> GlobalScopedPrivilege.fromXContent(n, p), APPLICATION);
+    }
+
+    private final Set<? extends GlobalScopedPrivilege> applicationPrivileges;
+
+    /**
+     * Constructs global privileges by bundling the set of application privileges.
+     * 
+     * @param applicationPrivileges
+     *            The privileges over applications.
+     */
+    public GlobalPrivileges(Collection<? extends GlobalScopedPrivilege> applicationPrivileges) {
+        if (applicationPrivileges == null || applicationPrivileges.isEmpty()) {
+            throw new IllegalArgumentException("Application privileges cannot be empty or null");
+        }
+        this.applicationPrivileges = Collections.unmodifiableSet(new HashSet<>(Objects.requireNonNull(applicationPrivileges)));
+        final Set<String> allScopes = this.applicationPrivileges.stream().map(p -> p.getScope()).collect(Collectors.toSet());
+        if (allScopes.size() != this.applicationPrivileges.size()) {
+            throw new IllegalArgumentException(
+                    "Application privileges have the same scope but the privileges differ. Only one privilege for any one scope is allowed.");
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.startObject(APPLICATION.getPreferredName());
+        for (final GlobalScopedPrivilege privilege : applicationPrivileges) {
+            builder.field(privilege.getScope(), privilege.getRaw());
+        }
+        builder.endObject();
+        return builder.endObject();
+    }
+
+    public static GlobalPrivileges fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public Set<? extends GlobalScopedPrivilege> getPrivileges() {
+        return applicationPrivileges;
```

Comment:
Thank you for the comprehensive explanation!

Really my though was that there's no way we can future proof this, the object will have to evolve in lock step with the JSON response (the parser is static, we cannot ignore unknwon fields, etc..). So I didn't gave it much though about how this "evolution" would look like. It would be breaking anyway, so who cares, design it for today.

But! I got a small idea from your `Map<String, Set<GlobalScopedPrivilege>>`, stay tuned!

Reply:
I've decided to use a `Set`. I think a map exposes internal implementation, the client should not be informed that we broken them down by category.